### PR TITLE
fix: migrate to 7.4

### PR DESCRIPTION
**Pull request by klaussilveira**

### CODE REVIEW

**Review by tamara-dietz — DISMISSED** [review]

**Comment by damian-af** [comment]
Hey @klaussilveira, now with the typed properties we need to take care of this:
imagen

**Comment by klaussilveira** [comment]
Where is that error happening?

**Comment by damian-af** [comment]
I fetched the branch in a simple project I created to test this out, I got that when running it.

File: src/Adapter/PhpredisAdapter.php [hunk 1]
```diff
@@ -40,13 +29,16 @@ public function __construct(array $config = [], bool $lazy = true)
 
     protected function getClient(): Redis
     {
-        if (!$this->client instanceof Redis) {
-            $this->createClient($this->config);
+        if (!$this->client) {
```
Comment (anesed) [inline src/Adapter/PhpredisAdapter.php:32]:
```suggestion
        if (!isset($this->client)) {
```
Would probably fix the error reported by @damian-af . We could also go for a default `null` value but, personally, I prefer the `isset` approach

Comment (damian-af) [reply src/Adapter/PhpredisAdapter.php:32]: Just checked and yes, that seems to work.

Comment (klaussilveira) [reply src/Adapter/PhpredisAdapter.php:32]: That seems to be a failure of phpstan to understand that `createClient` causes a side-effect of `$this->client` being defined. It's not so good for people too. I'm going to make `createClient` return a pointer to the instance just created so it is clearer.

**Comment by klaussilveira** [comment]
@damian-af :ping_pong:

**Comment by damian-af** [comment]
@klaussilveira still failing :(
imagen

File: src/Adapter/PhpredisAdapter.php (hunk 1 above)
Comment (damian-af) [inline src/Adapter/PhpredisAdapter.php]: We need to apply @anesed's suggestion.
imagen

Comment (damian-af) [reply src/Adapter/PhpredisAdapter.php]: I tried again and it is still failing.

**Review by damian-af — APPROVED** [review]

**Pull request merged**